webapp-django/crashstats/tests/test_sentrylib.py

```diff
@@ -0,0 +1,302 @@
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+"""Tests for Sentry event processing."""
+
+from copy import deepcopy
+from unittest import mock
+
+import pytest
+
+from crashstats.sentrylib import (SanitizeBreadcrumbs,
+                                  SanitizeHeaders,
+                                  SanitizePostData,
+                                  SanitizeQueryString,
+                                  SanitizeSQLQueryCrumb,
+                                  SentrySanitizer,
+                                  get_before_send)
+
+
+class TestSentrySanitizer:
+    """Tests for SentrySanitizer."""
+
+    @mock.patch('crashstats.sentrylib.markus', spec_set=('incr',))
+    def test_markus_exception(self, mock_markus):
+        """Exceptions are registered with markus."""
+
+        def test_sanitizer(event, hint):
+            raise RuntimeError('I raise errors.')
+
+        sanitizer = SentrySanitizer([test_sanitizer])
+        with pytest.raises(RuntimeError):
+            sanitizer({}, {})
+        mock_markus.incr.assert_called_once_with('before_send_exception')
+
+
+class TestSanitizeSQLQueryCrumb:
+    """Tests for SanitizeSQLQueryCrumb."""
+
+    CASES = {
+        # Select a user by email
+        'email': (
+            'SELECT "auth_user"."is_active" FROM "auth_user" WHERE'
+            ' UPPER("auth_user"."email"::text) = UPPER(\'username@example.com\')'),
+        # Find a user by username (usually same as email)
+        'username': (
+            'SELECT "auth_user"."is_active" FROM "auth_user" WHERE'
+            ' "auth_user"."username" = \'username\''),
+        # Set a user's password
+        'password': (
+            'UPDATE "auth_user" SET "password" = \'!unusable_pwd\' WHERE "auth_user"."id" = 2'),
+        # Update a session (includes session data, ID)
+        'session_data': (
+            'UPDATE "django_session" SET "session_data" = \'B64Data\', WHERE'
+            ' "django_session"."session_key" = \'a_session_key\''),
+        # Store a session (includes session ID, data)
+        'session_key': (
+            'INSERT INTO "django_session" ("session_key", "session_data") VALUES'
+            ' (\'a_session_key\', \'B64SessionData\'))'),
+        # Load data for an API token
+        'tokens_token.key': (
+            'SELECT "tokens_token"."id", "tokens_token"."user_id",'
+            ' "tokens_token"."key", "tokens_token"."expires",'),
+        # Update last login
+        'auth_user.id': (
+            'UPDATE \"auth_user\" SET \"last_login\" ='
+            ' datetime.datetime(2019, 5, 3, 18, 7, 29, 852498)'
+            ' WHERE \"auth_user\".\"id\" = 123'),
+    }
+
+    @pytest.mark.parametrize(
+        'keyword,sql', CASES.items(),
```

Comment:
I've been putting spaces between the arguments. So then this would be:

```
'keyword, sql'
```

For this one, it's not a big deal, but for larger argument lists, it's a lot easier to read.